kusd/validator/sync_waiter.go

```diff
@@ -0,0 +1,24 @@
+package validator
+
+import (
+	"github.com/kowala-tech/kUSD/event"
+	"github.com/kowala-tech/kUSD/kusd/downloader"
+)
+
+// SyncWaiter subscribes to DoneEvent and FailedEvent on a event.TypeMux
+// and when one of those events is received call a user function.
+// Example usage to delay Validator start after block sync has finished
+func SyncWaiter(eventMux *event.TypeMux, callback func(), callbackError func()) {
+	events := eventMux.Subscribe(downloader.DoneEvent{}, downloader.FailedEvent{})
+	defer events.Unsubscribe()
```

Comment:
Wouldn't it be better to unsubscribe before calling the callbacks and not after? We don't know how much time will the callback take, so I feel it's a bit more efficient to unsubscribe before calling them

Reply:
agree, wasn't sure, this was previous behaviour.